File: android/src/main/java/ws/dyt/plugin/umeng/analytics/log/LLog.java
package ws.dyt.plugin.umeng.analytics.log;

import android.util.Log;

/**
 * Created by yangxiaowei 2018/05/27
 */
public class LLog {

    private volatile static boolean debug = true;
    public static void setDebug(boolean debug) {
        LLog.debug = debug;
    }

    public static void d(String tag, String message) {

        if (debug) {
            Log.d(tag, message);
        }
    }

    public static void i(String tag, String message) {

        if (debug) {
            Log.i(tag, message);
        }
    }

    public static void w(String tag, String message) {

        if (debug) {
            Log.w(tag, message);
        }
    }

    public static void e(String tag, String message) {

        if (debug) {
            Log.e(tag, message);
        }
    }
}

File: lib/plugin/analytics/LLog.dart
//Created by yangxiaowei at 2018/06/07
class LLog {
    static bool _debug = true;
 
    static void setDebug(bool debug) {
        LLog._debug = debug;
    }

    static bool get debug => _debug;

    static void d(String tag, String message) {

        if (_debug) {
            _print('D', tag, message);
            
        }
    }

    static void i(String tag, String message) {

        if (_debug) {
            _print('I', tag, message);
        }
    }

    static void w(String tag, String message) {

        if (_debug) {
            _print('W', tag, message);
        }
    }

    static void e(String tag, String message) {

        if (_debug) {
            _print('E', tag, message);
        }
    }

    static void _print(String prefix, String tag, String message) {
        print('$prefix/$tag: $message');
    }
}